pyproject.toml

```diff
@@ -0,0 +1,7 @@
+[build-system]
+requires = [
+    "setuptools",
```

Comment:
You're right - there will be a minimum supported version of `setuptools`.  I actually don't know what it is, though, which is why I didn't put one.  That's not a _good_ reason by any stretch, but it is why it's the case.

A _very_ quick scan of the [setuptools history](https://setuptools.readthedocs.io/en/latest/history.html) lower-bounds the [newest required version to 43.0](https://setuptools.readthedocs.io/en/latest/history.html#v43-0-0) because we don't have `pyproject.toml` specified in `MANIFEST.in`, but it needs to be included for sdists to build correctly.  I may have missed a necessary change in a more recent version, though.  At any rate, if we added it explicitly, we could probably use setuptools all the way back to 36.6 (which added the PEP 517 build mechanism).  If that's the case, we don't really need to specify the dependency after; `setuptools.build_meta` only exists in 36.6, and pip knows that, but I'll check to see if it builds with 36.6 first.

Reply:
Excellent. To tests it on my side I  pulled from this `PR` into my `CuPy` environment and it installed in development mode with no hicups.